### PR TITLE
Fix #91 fetch post formats

### LINKED ISSUE #91: Implement FETCH_POST_FORMATS action in the SiteStore
And store the array for post formats in SiteModel

### PR DESCRIPTION
**Pull request by maxme**
Fix #91 fetch post formats -- cc @aforcier

### CODE REVIEW

File: fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java [hunk 1]
```diff
@@ -495,9 +517,27 @@ public void onAction(org.wordpress.android.fluxc.annotations.action.Action actio
...
+        } else if (actionType == SiteAction.FETCH_POST_FORMATS) {
+            fetchPostFormats((SiteModel) action.getPayload());
+        } else if (actionType == SiteAction.FETCHED_POST_FORMATS) {
+            FetchedPostFormatsPayload payload = (FetchedPostFormatsPayload) action.getPayload();
+            updatePostFormats(payload.site, payload.postFormats);
+        }
+    }
+
+    private void fetchPostFormats(SiteModel site) {
+        if (site.isWPCom()) {
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java]: Shouldn't we also check for `site.isJetpack()` here?

Comment (maxme) [inline fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java]: `isWPCom()` is true for Jetpack sites and I think it should stay like this. A more explicit name could be `isSiteAccessedViaWPComRestApi()`. `isJetpack()` would be used for in app stuff I guess, like enabling Jetpack options (stats) or enable some Jetpack-only features (Plugin management?).

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.java [hunk 2]
```diff
@@ -134,6 +138,36 @@ public void onErrorResponse(VolleyError error) {
         ));
     }
 
+    public void pullPostFormats(@NonNull final SiteModel site) {
+        final WPComGsonRequest<PostFormatsResponse> request = new WPComGsonRequest<>(Method.GET,
+                WPCOMREST.sites.site(site.getSiteId()).post_formats.getEndpoint(), null, PostFormatsResponse.class,
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/site/SiteRestClient.java]: This should be calling `post_formats.getUrlV1_1()`, not `getEndpoint()`.

File: example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestWPCOM.java [hunk 3]
```diff
@@ -75,6 +80,37 @@ public void testWPCOMSiteFetchAndLogout() throws InterruptedException {
...
+
+        // Fetch post formats
+        mDispatcher.dispatch(SiteActionBuilder.newFetchPostFormatsAction(firstSite));
+        mExpectedEvent = TEST_EVENTS.POST_FORMATS_CHANGED;
+        mCountDownLatch = new CountDownLatch(1);
+        assertEquals(true, mCountDownLatch.await(TestUtils.DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS));
+
+        // Test fetched Post Formats
+        List<PostFormatModel> postFormats = mSiteStore.getPostFormats(firstSite);
+        assertEquals(6, postFormats.size());
```
Comment (aforcier) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestWPCOM.java]: I'm getting 7 post formats from my .com site:
```
{"aside":"Aside","image":"Image","video":"Video","audio":"Audio","quote":"Quote","link":"Link","gallery":"Gallery"}}
```

Comment (maxme) [inline example/src/androidTest/java/org/wordpress/android/fluxc/release/ReleaseStack_SiteTestWPCOM.java]: I think that depends of the site's theme, can't set a hard number here :( `!= 0` should be good enough.

File: fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPCOMREST.java [hunk 4]
```diff
@@ -69,6 +69,15 @@ private PostEndpoint(String previousEndpoint, long postId) {
                     }
                 }
             }
+
+            @Endpoint("/sites/$site/post-formats/")
+            public final PostFormatsEndpoint post_formats = new PostFormatsEndpoint(getEndpoint());
```
Comment (aforcier) [inline fluxc/src/main/java/org/wordpress/android/fluxc/network/rest/wpcom/WPCOMREST.java]: If we don't plan on supporting any sub-endpoints anytime soon (`.../post-formats/new/`?), we can simplify this to `new WPComEndpoint(getEndpoint() + "post-formats/")` and drop `PostFormatsEndpoint` (see, e.g., the `/users/new/` endpoint below).

**Comment by aforcier** [comment]
:shipit:

**Pull request merged**